msr/msre/multiphysics_core_model/steady_state/th.i

```diff
@@ -0,0 +1,720 @@
+    porous_medium_treatment = true
+    add_energy_equation = true
+    gravity = '0.0 -9.81 0.0'
+
+    # Variable naming
+    velocity_variable = 'vel_x vel_y'
+    pressure_variable = 'pressure'
+    fluid_temperature_variable = 'T_fluid'
+
+    # Numerical schemes
+    pressure_face_interpolation = average
+    momentum_advection_interpolation = upwind
+    mass_advection_interpolation = upwind
+    energy_advection_interpolation = upwind
+    velocity_interpolation = rc
+
+    # Porous & Friction treatement
+    use_friction_correction = true
+    friction_types = 'darcy forchheimer'
+    friction_coeffs = 'Darcy_coefficient Forchheimer_coefficient'
+    consistent_scaling = 100.0
+    porosity_smoothing_layers = 2
+    turbulence_handling = 'mixing-length'
+
+    # fluid properties
+    density = 'rho'
+    dynamic_viscosity = 'mu'
+    thermal_conductivity = 'kappa'
+    specific_heat = 'cp'
+
+    # Energy source-sink
+    external_heat_source = 'power_density'
+
+    # boundary conditions
+    wall_boundaries = 'left      top      bottom   right    loop_boundary '
+    momentum_wall_types = 'symmetry  slip     noslip   noslip   noslip'
+    energy_wall_types = 'heatflux  heatflux heatflux heatflux heatflux'
+    energy_wall_function = '0        0        0        0        0'
+
+    pin_pressure = true
+    pinned_pressure_value = ${p_outlet}
+    pinned_pressure_type = average-uo
+
+    # passive scalar -- solved separetely to integrate porosity jumps
+    add_scalar_equation = false
+
+    #Scaling -- used mainly for nonlinear solves
+    momentum_scaling = 1e-3
+    mass_scaling = 10
+  []
+[]
+
+[FVKernels]
+  # Extra kernels for the thermal-hydraulics solve in the fluid
+  [pump_x]
+    type = INSFVBodyForce
+    variable = vel_x
+    functor = '${pump_force}'
+    block = 'pump'
+    momentum_component = 'x'
+  []
+  [pump_y]
+    type = INSFVBodyForce
+    variable = vel_y
+    functor = '${pump_force}'
+    block = 'pump'
+    momentum_component = 'y'
+  []
+  [convection_fluid_hx]
+    type = NSFVEnergyAmbientConvection
+    variable = T_fluid
+    T_ambient = ${T_inlet}
+    alpha = ${vol_hx}
+    block = 'pump'
+  []
+
+  # Kernels for solve in the solid blocks
+  [heat_time_solid]
+    type = INSFVEnergyTimeDerivative
+    variable = T_solid
+    cp = ${cp_steel}
+    rho = ${rho_steel}
+  []
+  [heat_diffusion_solid]
+    type = FVDiffusion
+    variable = T_solid
+    coeff = ${k_steel}
+  []
+  [convection_core]
+    type = PINSFVEnergyAmbientConvection
+    variable = T_solid
+    T_fluid = T_fluid
+    T_solid = T_solid
+    is_solid = true
+    h_solid_fluid = ${bulk_hx}
+    block = 'core'
+  []
+  [convection_core_completmeent]
+    type = PINSFVEnergyAmbientConvection
+    variable = T_fluid
+    T_fluid = T_fluid
+    T_solid = T_solid
+    is_solid = false
+    h_solid_fluid = ${bulk_hx}
+    block = 'core'
+  []
+
+  # Kernels for solve of delayed neutron precursor transport
```

Comment:
why are we not using the action for precursors?

Reply:
This was done explicitly so that advection to be modified to account for precursors jump due to porosity.